pkg/controller/control.go

```diff
@@ -213,6 +213,14 @@ func (c *Controller) Expand(size int64) error {
 	go func(size int64) {
 		expanded := false
 		defer func() {
+			// Frontend expansion involves in the iSCSI session rescanning, which will wait for the in-fly io requests complete.
+			// Hence there will be a deadlock once we use the lock to protect this frontend expansion.
+			if c.frontend != nil {
+				if err := c.frontend.Expand(size); err != nil {
+					logrus.WithError(err).Error("Failed to expand the frontend")
+					expanded = false
+				}
+			}
```

Comment:
Any potential side effect if no lock?

Reply:
cc @derekbit

Reply:
The frontend expansion consists of 2 parts:
1. The target LUN expansion. We just need to update the field `size` for the LUN. No data flow gets involved. This should be fine.
2. The iSCSI initiator refresh/rescan. The iSCSI should handle the data flow by itself rather than requiring the vendor/users to control it. And as I mentioned, adding the engine lock for this would lead to the deadlock issue.

Hence I don't think we need a lock here.

Reply:
Clarified, all good.